truffle/src/main/java/org/jruby/truffle/nodes/ext/BigDecimalNodes.java

```diff
@@ -178,175 +217,327 @@ public static boolean isNan(RubyBasicObject value) {
             return getBigDecimalType(value) == Type.NAN;
         }
 
+        private void setupCreateBigDecimal() {
+            if (createBigDecimal == null)
+                createBigDecimal = insert(CreateBigDecimalNodeFactory.create(getContext(), getSourceSection(), null, null));
+        }
+
+        protected RubyBasicObject createBigDecimal(VirtualFrame frame, Object value) {
+            setupCreateBigDecimal();
+            return createBigDecimal.executeCreate(frame, value);
+        }
+
+        protected RubyBasicObject createBigDecimal(VirtualFrame frame, Object value, RubyBasicObject self) {
+            setupCreateBigDecimal();
+            return createBigDecimal.executeCreate(frame, value, self);
+        }
+
+        private void setupLimitCall() {
+            if (limitCall == null)
+                limitCall = insert(DispatchHeadNodeFactory.createMethodCall(getContext()));
```

Comment:
I think maybe they could do, but I prefer to be really explicit with transfers. We don't want to let work leak from one side of a transfer to another by trying to be to magic about it. I prefer the explicit clean break that the directive gives us.